frontend/src/components/Home.js

```diff
@@ -83,25 +83,36 @@ export default function Home() {
 
   useEffect(() => {
     let url = `${API_HOST}/leads?page=${page}&perpage=${perpage}`;
+    const n_pagesUrl = `${API_HOST}/leads/n_pages`;
```

Comment:
If you check the [swagger docs](https://cfd-partner-finder-api.xyz/apidocs/#/default/get_leads_n_pages), there is a `perpage` query parameter to `/leads/n_pages`. It should save you from redoing the math that is done on the backend.

```suggestion
    const n_pagesUrl = `${API_HOST}/leads/n_pages?perpage=${perpage}`;
```